sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -43,10 +44,18 @@ case class BasicWriteTaskStats(
     partitions: Seq[InternalRow],
     numFiles: Int,
     numBytes: Long,
-    numRows: Long)
+    numRows: Long,
+    partitionsStats: Map[InternalRow, BasicWritePartitionTaskStats]
```

Comment:
have you measured the perf overhead of this? It may significantly increase the binary size of task stats.

Reply:
I used `InternalRow` (instead of path) to try to keep this constrained, and  `BasicWritePartitionTaskStats` is limited to an int and 2 longs.  I'll plan on more additional performance tests.